File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        ans,n=0,len(nums)
        ps=list(accumulate(nums,operator.add))
        total_sum = ps[-1]
        for i in range(n-1): 
            if ps[i]>=total_sum -ps[i]:
                ans+=1
        return ans

        

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.cpp
#include <vector>
#include <numeric>
#include <functional>
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int ans = 0;
        int n = nums.size();
        vector<long long> ps(n);
        
        ps[0] = nums[0];
        for (int i = 1; i < n; ++i) {
            ps[i] = ps[i - 1] + nums[i];
        }
        
        long long total_sum = ps.back();
        
        for (int i = 0; i < n - 1; ++i) {
            if (ps[i] >= total_sum - ps[i]) {
                ans++;
            }
        }
        
        return ans;
    }
};